docs/spec/metrics/metric.json

```diff
@@ -0,0 +1,39 @@
+{
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "$id": "docs/spec/metrics/metric.json",
+    "type": "object",
+    "description": "Metric data captured by an APM agent",
+    "properties": {
+        "samples": {
+            "type": "object",
+            "description": "Sampled application metrics collected from the agent",
+            "regexProperties": true,
+            "patternProperties": {
+                "^[^*\"]*$": {
+                    "$ref": "sample.json",
+                    "maxLength": 1024
+                }
+            },
+            "additionalProperties": false
+        },
+        "tags": {
+            "type": ["object", "null"],
+            "description": "A flat mapping of user-defined tags with string values.",
+            "regexProperties": true,
+            "patternProperties": {
+                "^[^*\"]*$": {
+                    "type": "string",
```

Comment:
I'm not clear on what you're proposing here.  Does this mean we should support a payload like this?:
```
tags: {
  "foo": null
}
```

Reply:
Yes thats what we usually allow. The null values are removed during the server side processing, we are just less strict on the intake side. If this is not allowed, the whole request would fail.

Reply:
Ok, I can live with that, will update.